src/slurm_plugin/fleet_manager.py

```diff
@@ -47,9 +47,14 @@ def __hash__(self):
     @staticmethod
     def from_describe_instance_data(instance_info):
         try:
+            private_ip = instance_info["PrivateIpAddress"]
+            for network_interface in instance_info["NetworkInterfaces"]:
+                attachment = network_interface["Attachment"]
+                if attachment["DeviceIndex"] == 0 and attachment["NetworkCardIndex"] == 0:
+                    private_ip = network_interface["PrivateIpAddress"]
```

Comment:
Any reason not to break out of this loop if we find what we are looking for?

Reply:
No reason, we should break